crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -294,29 +311,53 @@ impl WasiRuntime {
                 Err(e) => {
                     let message = "unable to run module";
                     error!("{}: {:?}", message, e);
-                    status_sender
-                        .broadcast(Status::Terminated {
+                    send(
+                        status_sender.clone(),
+                        name.clone(),
+                        Status::Terminated {
                             failed: true,
                             message: message.into(),
                             timestamp: chrono::Utc::now(),
-                        })
-                        .expect("status should be able to send");
+                        },
+                        &mut cx,
+                    );
                     return Err(anyhow::anyhow!("{}: {}", message, e));
                 }
             };
 
             info!("module run complete");
-            status_sender
-                .broadcast(Status::Terminated {
+            send(
+                status_sender.clone(),
+                name,
+                Status::Terminated {
                     failed: false,
                     message: "Module run completed".into(),
                     timestamp: chrono::Utc::now(),
-                })
-                .expect("status should be able to send");
+                },
+                &mut cx,
+            );
             Ok(())
         });
         // Wait for the interrupt to be sent back to us
         let interrupt = rx.await?;
         Ok((interrupt, handle))
     }
 }
+
```

Comment:
I understand the need for this channel, but it seems to be implemented in very cumbersome way here. Am I missing something?

Reply:
We changed the internals from a `tokio::sync::watch` to a `tokio::sync::mpsc::Sender`. The former did not return a future on a `.broadcast`, but `.send` does. Because async calls like `await` cannot be used inside non-async threads (like the one spawned in `spawn_wasmtime`), thus born the cumbersome `send`.

In reality these are all gross hacks to get the tests passing again. Once that's done we can take a step back and remove some of this tech debt. If you have a suggestion how we could make this better in the meantime, we'd be happy to hear your feedback.

Reply:
Ah, I did not catch that the calling function was not Async. Makes sense.